python/docstring.cpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Neil Vaytet
+#include "docstring.h"
+
+Docstring &Docstring::description(const std::string &s, const bool append) {
+  if (append)
+    m_description += s;
+  else
+    m_description = s;
+  return *this;
+}
+
+Docstring &Docstring::raises(const std::string &s, const bool append) {
+  if (append)
+    m_raises += s;
+  else
+    m_raises = s;
+  return *this;
+}
+
+Docstring &Docstring::seealso(const std::string &s, const bool append) {
+  if (append)
+    m_seealso += s;
+  else
+    m_seealso = s;
+  return *this;
+}
+
+Docstring &Docstring::returns(const std::string &s, const bool append) {
+  if (append)
+    m_returns += s;
+  else
+    m_returns = s;
+  return *this;
+}
+
+Docstring &Docstring::rtype(const std::string &s, const bool append) {
+  if (append)
+    m_rtype += s;
+  else
+    m_rtype = s;
+  return *this;
+}
+
+Docstring &Docstring::param(const std::string &name, const std::string &about,
+                            const std::string &type) {
+  if (m_params.find(name) == m_params.end()) {
+    m_order.push_back(name);
+  }
+  m_params[name] = {about, type};
+  return *this;
+}
+
+void Docstring::clear() {
+  m_description.clear();
+  m_raises.clear();
+  m_seealso.clear();
+  m_returns.clear();
+  m_rtype.clear();
+  m_order.clear();
+  m_params.clear();
+  m_output.clear();
+}
```

Comment:
Not sure why this method is needed, it feels like a dangerous design (will break when someone adds a new field). What is wrong with using `doc = Docstring()`?

Reply:
Ok, will go with your suggestion. It will indeed break when someone adds a new field, and in fact this already happened ;-)
I managed to break it myself by adding a new field!
Thanks for the suggestion